tests/e2e/specs/backend/product-query/advanced-filters.test.ts

```diff
@@ -0,0 +1,160 @@
+			 */
+			await resetProductQueryBlockPage();
+			await ensureSidebarOpened();
+			await selectBlockByName( block.slug );
+			$productFiltersPanel = await findToolsPanelWithTitle(
+				'Advanced Filters'
+			);
+		} );
+
+		/**
+		 * Reset the content of Product Query Block page after this test suite
+		 * to avoid breaking other tests.
+		 */
+		afterAll( async () => {
+			await resetProductQueryBlockPage();
+		} );
+
+		describe( 'Sale Status', () => {
+			it( 'Sale status is disabled by default', async () => {
+				await expect( $productFiltersPanel ).not.toMatch(
+					'Show only products on sale'
+				);
+			} );
+
+			it( 'Can add and remove Sale Status filter', async () => {
+				await toggleAdvancedFilter( 'Sale status' );
+				await expect( $productFiltersPanel ).toMatch(
+					'Show only products on sale'
+				);
+				await toggleAdvancedFilter( 'Sale status' );
+				await expect( $productFiltersPanel ).not.toMatch(
+					'Show only products on sale'
+				);
+			} );
+
+			it( 'Editor preview shows correct products corresponding to the value `Show only products on sale`', async () => {
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+				await toggleAdvancedFilter( 'Sale status' );
+				await setCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength( saleCount );
+				await unsetCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+			} );
+
+			it( 'Works on the front end', async () => {
+				await toggleAdvancedFilter( 'Sale status' );
+				await setCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength( saleCount );
+				await saveOrPublish();
+				await shopper.block.goToBlockPage( block.name );
+				expect( await getFrontEndProducts() ).toHaveLength( saleCount );
+			} );
+		} );
+
+		describe( 'Stock Status', () => {
+			it( 'Stock status is enabled by default', async () => {
+				await expect( $productFiltersPanel ).toMatchElement(
+					SELECTORS.formTokenField.label,
+					{ text: 'Stock status' }
+				);
+			} );
+
+			it( 'Can add and remove Stock Status filter', async () => {
+				await toggleAdvancedFilter( 'Stock status' );
+				await expect( $productFiltersPanel ).not.toMatchElement(
+					SELECTORS.formTokenField.label,
+					{ text: 'Stock status' }
+				);
+				await toggleAdvancedFilter( 'Stock status' );
+				await expect( $productFiltersPanel ).toMatchElement(
+					SELECTORS.formTokenField.label,
+					{ text: 'Stock status' }
+				);
+			} );
+
+			it( 'All statuses are enabled by default', async () => {
+				await expect( $productFiltersPanel ).toMatch( 'In stock' );
+				await expect( $productFiltersPanel ).toMatch( 'Out of stock' );
+				await expect( $productFiltersPanel ).toMatch( 'On backorder' );
```

Comment:
Not a big deal, but should this also be perhaps related to the fixture data?

We could have a loop on something like `getFixtureProductsData( 'stock_statii' )` and make it more future proof?

Reply:
- Fixture doesn't have any information about all available statuses right now, we need to hard-code it if we want to get stock statuses from it.
- In the future, if WC Core changes the default stock statuses, we will need to update the fixture as well. To me, updating this test or updating the fixture is similar. Test should be updated as the WC and WC Blocks update.
- I checked WooCommerce REST API to see if there is any endpoint returning available stock status, but sadly there is none.
- We can centeralize the reference for available stock status options, but it should be a utility or a constants, not in the fixture data.